@here/harp-mapview/lib/Utils.ts

```diff
@@ -136,31 +147,26 @@ export namespace MapViewUtils {
             // FIXME: We cannot use mapView.tilt here b/c it does not reflect the latest camera
             // changes.
             // tslint:disable-next-line: deprecation
-            const tilt = extractCameraTilt(mapView.camera, mapView.projection);
+            const tilt = extractCameraTilt(camera, mapView.projection);
             const deltaTilt = tilt - maxTiltAngle;
             if (deltaTilt > 0) {
                 orbitFocusPoint(mapView, 0, deltaTilt, maxTiltAngle);
             }
         }
 
         // Get new target position after the zoom
-        const newTargetPosition = rayCastWorldCoordinates(
-            mapView,
-            targetPositionOnScreenXinNDC,
-            targetPositionOnScreenYinNDC
-        );
-
-        if (!targetPosition || !newTargetPosition) {
+        const newWorldTarget = rayCastWorldCoordinates(mapView, targetNDCx, targetNDCy, elevation);
+        if (!newWorldTarget) {
             return;
         }
 
         if (mapView.projection.type === ProjectionType.Planar) {
             // Calculate the difference and pan the map to maintain the map relative to the target
             // position.
-            targetPosition.sub(newTargetPosition);
-            panCameraAboveFlatMap(mapView, targetPosition.x, targetPosition.y);
+            worldTarget.sub(newWorldTarget);
+            panCameraAboveFlatMap(mapView, worldTarget.x, worldTarget.y);
         } else if (mapView.projection.type === ProjectionType.Spherical) {
-            panCameraAroundGlobe(mapView, targetPosition, newTargetPosition);
+            panCameraAroundGlobe(mapView, worldTarget, newWorldTarget);
```

Comment:
I see slight problem here, although not introduced by your change... What happens when we encounter huge elevations in between? Camera may fly though the geometry? Although assuming that raycast is made with ndc and eleveation we should never aim to point which is "hidden" with some bigger mountain. Have you done such testing?

Reply:
There a terrain-camera interaction ticket where we can handle all that:
https://saeljira.it.here.com/browse/HARP-5430

This change is meant to be a small fix that makes globe with terrain at least reasonably usable (right now it's completely broken).